### ISSUE TITLE
Error running Flutter 3 and amplify_datastore

### ISSUE THREAD

**Issue by veeramneni** [aws-amplify/amplify-flutter#1583]
### Description

Started a new flutter project with flutter 3 and amplify datastore - 0.4.5. Pub get imports fine but when running the project build is failing with

flutter/.pub-cache/hosted/pub.dartlang.org/amplify_datastore-0.4.5/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt: (519, 75): Object is not abstract and does not implement abstract member public abstract fun error(p0: String, p1: String?, p2: Any?): Unit defined in io.flutter.plugin.common.MethodChannel.Result

flutter/.pub-cache/hosted/pub.dartlang.org/amplify_datastore-0.4.5/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt: (531, 29): 'error' overrides nothing

I understand Flutter 3 is just out today but curious if there is an easy fix or we either need to wait for next amplify update or use older version of flutter.

Also other amplify libraries don't have this issue. Datastore is the only issue.

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [X] DataStore
- [ ] Storage

### Steps to Reproduce

1. Create new empty flutter project with most recent Flutter
2. Add amplify_flutter and amplify_datstore to pubspec.yaml and do pub get.
3. Run the project.

### Platforms

- [ ] iOS
- [X] Android

### Environment

```bash
[✓] Flutter (Channel stable, 3.0.0, on macOS 12.3.1 21E258 darwin-arm, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.67.0)
[✓] Connected device (3 available)
[✓] HTTP Host Availability
```

### Dependencies

```bash
- amplify_analytics_pinpoint 0.4.5 [amplify_analytics_plugin_interface amplify_analytics_pinpoint_android amplify_analytics_pinpoint_ios amplify_core flutter plugin_platform_interface]
- amplify_api 0.4.5 [amplify_api_plugin_interface amplify_core collection flutter meta plugin_platform_interface]
- amplify_auth_cognito 0.4.5 [flutter amplify_auth_plugin_interface amplify_core amplify_auth_cognito_android amplify_auth_cognito_ios collection plugin_platform_interface]
- amplify_datastore 0.4.5 [flutter amplify_datastore_plugin_interface amplify_core plugin_platform_interface meta collection async]
- amplify_flutter 0.4.5 [amplify_analytics_plugin_interface amplify_api_plugin_interface amplify_auth_plugin_interface amplify_core amplify_datastore_plugin_interface amplify_storage_plugin_interface collection flutter json_annotation meta plugin_platform_interface]
- cupertino_icons 1.0.4
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_login 3.2.0 [flutter font_awesome_flutter provider another_transformer_page_view another_flushbar flutter_signin_button quiver url_launcher]
- font_awesome_flutter 9.2.0 [flutter]
```

### Device

Pixel 3

### OS

N/A

### CLI Version

8.1.0

**Comment by HuiSF** [comment]
Thanks for opening this issue @veeramneni .
A PR has been created now to fix this issue.
https://github.com/aws-amplify/amplify-flutter/pull/1580

Before the fix to be released, I recommend you using Flutter < 2.17 for development.

**Comment by dnys1** [comment]
This should be fixed in ^0.5.0
